circuits/cpp/src/aztec3/circuits/abis/private_kernel/private_call_data.hpp

```diff
@@ -129,4 +129,12 @@ template <typename NCT> std::ostream& operator<<(std::ostream& os, PrivateCallDa
               << "acir_hash: " << obj.acir_hash << "\n";
 }
 
-}  // namespace aztec3::circuits::abis::private_kernel
\ No newline at end of file
+}  // namespace aztec3::circuits::abis::private_kernel
+
+// specialize the name in msgpack schema generation
+// consumed by the typescript schema compiler, helps disambiguate templates
```

Comment:
I don't think I understand this - Is this needed for every cbind template?

Reply:
This just sets the typescript interface name, which is not easy to figure out otherwise as we pass compound objects to the template. I could possibly be swayed to analyze the big templated types in TS, but currently C++ helps